### ISSUE TITLE
Semicolon rule misses return statements

### ISSUE THREAD

**Issue by nzakas** [eslint/eslint#616]
The following will not be flagged by the `semi` rule:

``` js
function foo() { 
    return []
}
```

JSHint and JSLint both flag the `return` statement as missing a semicolon, so should we.
